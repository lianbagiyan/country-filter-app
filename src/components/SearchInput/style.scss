.filter-input {
  width: 400px;
  height: 30px;
  padding-left: 10px;

  &:focus {
    outline: none;
  }
}
