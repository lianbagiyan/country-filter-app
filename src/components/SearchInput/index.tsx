import React from 'react';
import { SearchInputProps } from "../../types/types";
import './style.scss';

const SearchInput: React.FC<SearchInputProps> = ({ value, onChange }) => {
    return (
        <input
            type="text"
            className="filter-input"
            placeholder="Filter by country code"
            value={value}
            onChange={(e) => onChange(e.target.value)}
        />
    );
};

export default SearchInput;
