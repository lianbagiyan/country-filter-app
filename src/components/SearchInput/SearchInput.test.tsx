import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import SearchInput from './index';

describe('SearchInput Component', () => {
  test('renders correctly', () => {
    const onChange = jest.fn();
    render(<SearchInput value="" onChange={onChange} />);

    const inputElement = screen.getByPlaceholderText('Filter by country code');
    expect(inputElement).toBeInTheDocument();
  });

  test('calls onChange prop when input value changes', () => {
    const onChange = jest.fn();
    render(<SearchInput value="" onChange={onChange} />);

    const inputElement = screen.getByPlaceholderText('Filter by country code');
    fireEvent.change(inputElement, { target: { value: 'MG' } });

    expect(onChange).toHaveBeenCalledWith('MG');
  });

  test('clears input correctly', () => {
    const onChange = jest.fn();
    render(<SearchInput value="MG" onChange={onChange} />);

    const inputElement = screen.getByPlaceholderText('Filter by country code');
    fireEvent.change(inputElement, { target: { value: '' } });

    expect(onChange).toHaveBeenCalledWith('');
  });

  test('matches snapshot', () => {
    const { container } = render(<SearchInput value="" onChange={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
