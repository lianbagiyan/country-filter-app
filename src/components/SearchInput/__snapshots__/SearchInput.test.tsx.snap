// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component matches snapshot 1`] = `
<div>
  <input
    class="filter-input"
    placeholder="Filter by country code"
    type="text"
    value=""
  />
</div>
`;
